source/extensions/resource_monitors/downstream_connections/downstream_connections_monitor.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "envoy/extensions/resource_monitors/downstream_connections/v3/downstream_connections.pb.h"
+#include "envoy/server/proactive_resource_monitor.h"
+
+#include "source/common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace DownstreamConnections {
+
+class ActiveDownstreamConnectionsResourceMonitor : public Server::ProactiveResourceMonitor {
+public:
+  ActiveDownstreamConnectionsResourceMonitor(
+      const envoy::extensions::resource_monitors::downstream_connections::v3::
+          DownstreamConnectionsConfig& config);
+
+  bool tryAllocateResource(int64_t increment) override;
+
+  bool tryDeallocateResource(int64_t decrement) override;
+
+  int64_t currentResourceUsage() const override;
+  int64_t maxResourceUsage() const override;
+
+protected:
+  uint64_t max_;
```

Comment:
Thoughts on perhaps changing this, and where it's set in the proto to int64_t instead? The pgv rule can enforce this value is > 0 (if that was the concern) Currently, there are values this could be set to where the static cast is unsafe